packages/extension/src/background/activeTabs.ts

```diff
@@ -3,32 +3,65 @@ import browser from "webextension-polyfill"
 import { sendMessage } from "../shared/messages"
 import { MessageType } from "../shared/MessageType"
 
-const activeTabs = new Set<number>()
+type Tab = { id: number; host: string }
+const activeTabs: Tab[] = []
 
-export function addTab(tabId?: number) {
-  if (tabId !== undefined && !hasTab(tabId)) {
-    activeTabs.add(tabId)
+export function addTab(tab: Tab) {
+  if (tab.id !== undefined && !hasTab(tab.id)) {
+    activeTabs.push(tab)
+  }
+}
+
+export function removeTab(tabId?: number) {
+  if (tabId !== undefined && hasTab(tabId)) {
+    activeTabs.splice(
+      activeTabs.findIndex((tab) => tab.id === tabId),
+      1,
+    )
   }
 }
 
 export function hasTab(tabId?: number) {
   if (tabId === undefined) {
     return false
   }
-  return activeTabs.has(tabId)
+  return activeTabs.some((tab) => tab.id === tabId)
 }
 
 browser.tabs.onRemoved.addListener((tabId) => {
```

Comment:
Would:

```suggestion
browser.tabs.onRemoved.addListener(removeTab)
```

be more graceful?